lib/model/autocomplete.dart

```diff
@@ -244,11 +246,116 @@ class MentionAutocompleteView extends ChangeNotifier {
     notifyListeners();
   }
 
+  List<User>? _sortedUsers;
+
+  int compareByDms(User userA, User userB) {
+    final aRecencyIndex = store.recentDmConversationsView.getRecencyIndex(userA.userId);
+    final bRecencyIndex = store.recentDmConversationsView.getRecencyIndex(userB.userId);
+
+    if (aRecencyIndex > bRecencyIndex) {
+      return -1;
+    } else if (bRecencyIndex > aRecencyIndex) {
+      return 1;
+    }
+
+    final aIsPartner = store.recentDmConversationsView.isPartner(userA.userId);
+    final bIsPartner = store.recentDmConversationsView.isPartner(userB.userId);
```

Comment:
I wonder if this has any effect in our case! In the `recent_dm_conversations.dart` file, the `final Map<int, int> _dmRecencyData = {};` line in fact keeps track of all the DM users besides the latest message ID. So theoretically, if there is no recent message exchanged, it means we never had a conversation with that user!